src/ray/raylet/worker_pool.cc

```diff
@@ -91,11 +90,13 @@ void WorkerPool::RegisterWorker(std::shared_ptr<Worker> worker) {
   auto pid = worker->Pid();
   RAY_LOG(DEBUG) << "Registering worker with pid " << pid;
   registered_workers_.push_back(std::move(worker));
+
   auto it = starting_worker_processes_.find(pid);
-  RAY_CHECK(it != starting_worker_processes_.end());
-  it->second--;
-  if (it->second == 0) {
-    starting_worker_processes_.erase(it);
+  if (it != starting_worker_processes_.end()) {
```

Comment:
Why do we need this change?

Reply:
I have explained it in this [comment](https://github.com/ant-tech-alliance/ray/pull/54#discussion-diff-204663832L98).
@raulchen  cc